qcodes/station.py

```diff
@@ -76,6 +77,10 @@ class ValidationWarning(Warning):
     pass
 
 
+class StationConfig(UserDict):
```

Comment:
I'd vote for making first a quick PR that fixes this (or just in this PR), and then we won't need to have the ugly union in the station.components
```suggestion
class StationConfig(Metadatable):
```

Reply:
@Dominik-Vogel Do you have any opinion on this? I am happy to make the change but I may be missing the context

Reply:
I would more lean towards removing config from components and having it as a specific separate entry in the station and update snapshot_base to incorporate it

Reply:
> I would more lean towards removing config from components and having it as a specific separate entry in the station and update snapshot_base to incorporate it

agree, let's not pollute "components".

Reply:
I am also all for that suggestion.